src/libcrun/criu.c

```diff
@@ -232,6 +251,54 @@ libcrun_container_restore_linux_criu (libcrun_container_status_t *status,
 
   criu_set_images_dir_fd (image_fd);
 
+  /* Load descriptors.json to tell CRIU where those FDs should be connected to. */
+  {
+    cleanup_free char *descriptors_path = NULL;
+    cleanup_free char *buffer = NULL;
+    char err_buffer[256];
+    yajl_val tree;
+
+    xasprintf (&descriptors_path, "%s/%s", cr_options->image_path,
+               DESCRIPTORS_FILENAME);
+    ret = read_all_file (descriptors_path, &buffer, NULL, err);
+    if (UNLIKELY (ret < 0))
+      return ret;
+
+    status->external_descriptors = xstrdup (buffer);
```

Comment:
not blocking issue, but will it ever be freed?

Reply:
No, not really. In linux.c it will be passed to container->private_data->external_descriptors and I see no place where this struct is cleaned up.

Reply:
it is fine, anyway it is just allocated once, and anyway the error already existed.